[FME] React SDK: update 'Server-Side Rendering' section with the new `initialRolloutPlan` feature (#11579)
harness/developer-hub: docs/feature-management-experimentation/20-sdks-and-infrastructure/client-side-sdks/react-sdk.md

</TabItem>
</Tabs>

#### SDK ready from cache on initial render

> If the configuration object is provided with an `initialRolloutPlan` option, the SDK clients will be **ready from cache** immediately, including the initial component tree render.

Reviewer: ```suggestion
When you provide an `initialRolloutPlan` in the configuration, the SDK clients becomes **ready from cache** immediately, including the initial component tree render.
```

> At this point, the `isReadyFromCache` status property is `true` and the treatments retrieved using `useSplitTreatments` are based on the feature flag definitions provided in the `initialRolloutPlan` option.

Reviewer: ```suggestion
In this state, the `isReadyFromCache` status property is `true` and `useSplitTreatments` returns treatments based on the feature flag definitions in the `initialRolloutPlan`.
```

> The flow to get the rollout plan snapshot and pass it down to the SDK via the `initialRolloutPlan` configuration option consists on the following steps:

Reviewer: ```suggestion
To get the rollout plan snapshot and pass it to the SDK via `initialRolloutPlan`, follow these steps:
```

> 1) Extract the rollout plan snapshot using the `getRolloutPlan` method of the Node.js SDK factory.
> 2) Create the SDK configuration object with the `initialRolloutPlan` option.
> 3) Serialize and inject the SDK configuration object in the rendered HTML. You can do this manually (see "Vanilla SSR" example below), but Web frameworks generally provide a more convenient way to do this, for example using the `getServerSideProps` function in Next.js (see "Next.js using Pages Router" example below).
> 4) Pass the SDK configuration object to the `SplitFactoryProvider` component.

Reviewer: ```suggestion
1. Extract the rollout plan snapshot using the Node.js SDK factory's `getRolloutPlan` method.
1. Create the SDK configuration object with the `initialRolloutPlan` option.
1. Serialize and inject the SDK configuration object into the rendered HTML. You can do this manually (see the "Vanilla SSR" example below) or use framework-specific methods, like Next.js's `getServerSideProps` function (see the "Next.js using Pages Router" example below).
1. Pass the SDK configuration object to the `SplitFactoryProvider` component.
```
